plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -155,19 +156,23 @@ public ConnectorPageSource createPageSource(
         ColumnMappingMode columnMappingMode = getColumnMappingMode(table.getMetadataEntry(), table.getProtocolEntry());
         Optional<List<String>> partitionValues = Optional.empty();
         if (deltaLakeColumns.stream().anyMatch(column -> column.getBaseColumnName().equals(ROW_ID_COLUMN_NAME))) {
+            // using ArrayList because partition values can be null
             partitionValues = Optional.of(new ArrayList<>());
-            for (DeltaLakeColumnMetadata column : extractSchema(table.getMetadataEntry(), table.getProtocolEntry(), typeManager)) {
+            List<DeltaLakeColumnMetadata> columnsMetadata = extractSchema(table.getMetadataEntry(), table.getProtocolEntry(), typeManager);
+            for (String partitionColumnName : table.getMetadataEntry().getOriginalPartitionColumns()) {
+                DeltaLakeColumnMetadata partitionColumn = columnsMetadata.stream()
```

Comment:
This should be okay, but doesn't really look that efficient (especially with more partition columns). I'm wondering if it's better to store `columnsMetadata` in `Map<String, DeltaLakeColumnMetadata>` so then you can just do `DeltaLakeColumnMetadata partitionColumn = columnMetadata.get(partitionColumnName)`.